src/Microsoft.ML.Transforms/TransformsStatic.cs

```diff
@@ -117,4 +118,185 @@ public Reconciler(bool subMean, bool useStdDev, float scale)
+        /// <param name="slotsInOutput">The maximum number of slots to preserve in the output. The number of slots to preserve is taken across all input columns.</param>
+        /// <param name="numBins">Max number of bins used to approximate mutual information between each input column and the label column. Power of 2 recommended.</param>
+        public static Vector<bool> SelectFeaturesBasedOnMutualInformation(
+            this Vector<bool> input,
+            Scalar<bool> labelColumn,
+            int slotsInOutput = MutualInformationFeatureSelectingEstimator.Defaults.SlotsInOutput,
+            int numBins = MutualInformationFeatureSelectingEstimator.Defaults.NumBins) => new OutPipelineColumn<bool>(input, labelColumn, slotsInOutput, numBins);
+
+        /// <include file='doc.xml' path='doc/members/member[@name="MutualInformationFeatureSelection"]/*' />
+        /// <param name="input">Name of the input column.</param>
+        /// <param name="labelColumn">Name of the column to use for labels.</param>
+        /// <param name="slotsInOutput">The maximum number of slots to preserve in the output. The number of slots to preserve is taken across all input columns.</param>
+        /// <param name="numBins">Max number of bins used to approximate mutual information between each input column and the label column. Power of 2 recommended.</param>
+        public static Vector<bool> SelectFeaturesBasedOnMutualInformation(
+            this Vector<bool> input,
+            Scalar<float> labelColumn,
+            int slotsInOutput = MutualInformationFeatureSelectingEstimator.Defaults.SlotsInOutput,
+            int numBins = MutualInformationFeatureSelectingEstimator.Defaults.NumBins) => new OutPipelineColumn<bool>(input, labelColumn, slotsInOutput, numBins);
+    }
+
+    /// <summary>
+    /// Extensions for statically typed <see cref="CountFeatureSelectorExtensions"/>.
+    /// </summary>
+    public static class CountFeatureSelectorExtensions
+    {
+        private sealed class OutPipelineColumn<T> : Vector<T>
+        {
+            public readonly Vector<T> Input;
+
+            public OutPipelineColumn(Vector<T> input, long count)
+                : base(new Reconciler<T>(count), input)
+            {
+                Input = input;
+            }
+        }
+
+        private sealed class Reconciler<T> : EstimatorReconciler
+        {
+            private readonly long _count;
+
+            public Reconciler(long count)
+            {
+                _count = count;
+            }
+
+            public override IEstimator<ITransformer> Reconcile(IHostEnvironment env,
+                PipelineColumn[] toOutput,
+                IReadOnlyDictionary<PipelineColumn, string> inputNames,
+                IReadOnlyDictionary<PipelineColumn, string> outputNames,
+                IReadOnlyCollection<string> usedNames)
+            {
+                env.Assert(toOutput.Length == 1);
```

Comment:
>env.Assert(toOutput.Length == 1); [](start = 16, length = 33)

why we allow only one column?

Reply:
This came from the previous code (estimator wrapper), I don't think it applies to the new code any longer, so I removed it.

---
In reply to: [235534881](https://github.com/dotnet/machinelearning/pull/1683#discussion_r235534881) [](ancestors = 235534881)